app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,73 @@
+class TweetsController < ApplicationController
+
+	get '/tweets' do 
+		if logged_in?
+		@tweets = Tweet.all
+		erb :'tweets/index'
+		else 
+			redirect '/login'
+		end
+	end
+
+	get '/tweets/new' do 
+		if logged_in?
+			erb :'tweets/new'
+		else
+			redirect '/login'
+		end
+	end
+
+	post '/tweets' do
+		if params[:content].blank?
+			redirect "/tweets/new"
+		else
+		@tweet = current_user.tweets.create(content: params[:content])
+		redirect "/tweets/#{@tweet.id}"
+		end
+	end
+
+	get '/tweets/:id' do 
+		if logged_in?
+		@tweet = Tweet.find_by_id(params[:id])
+		erb :'tweets/show'
+		else
+			redirect '/login'
+		end
+	end
+
+	get '/tweets/:id/edit' do
+		if logged_in?
+			@tweet = Tweet.find_by_id(params[:id])
+			if current_user.id == @tweet.user_id
+		 		erb :'tweets/edit' 
+		 	else
+		 		redirect to '/tweets'
+		 	end
+		
+		else
+			redirect '/login'
+		end
+	end
+
+	post '/tweets/:id' do
+		if params[:content].blank?
+			redirect "/tweets/#{params[:id]}/edit"
+		else
+			@tweet = Tweet.find_by_id(params[:id])
+			@tweet.update(content: params[:content])
+			redirect "/tweets/#{@tweet.id}"
+		end		
+	end
+
+	post '/tweets/:id/delete' do 
+		if logged_in?
+			@tweet = Tweet.find_by_id(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.